Package:     capi4hylafax
Subject:     05 fix fprintf params
Author:      Moritz Muehlenhoff <jmm at debian.org>
Bug-Debian:  http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=653539
Forwarded:   no
Last-Update: 2012-06-12

Fix various code bits to be compatible with -Wformat-security.
---

diff -urN s05/src/faxrecv/faxrecv.cpp s06/src/faxrecv/faxrecv.cpp
--- s05/src/faxrecv/faxrecv.cpp	2012-06-12 19:54:35.844950804 +0200
+++ s06/src/faxrecv/faxrecv.cpp	2012-06-12 20:42:18.397101056 +0200
@@ -443,7 +443,7 @@
     sprintf (timebuf + strlen(timebuf), ".%02lu: [%5d]: ", tv.tv_usec / 10000, getpid());
 
     if (hJobFile) {
-        fprintf (hJobFile, timebuf);
+        fprintf (hJobFile, "%s", timebuf);
         vfprintf (hJobFile, text, params);
         va_end (params);
         va_start (params, text);
diff -urN s05/src/faxrecv/recvmain.cpp s06/src/faxrecv/recvmain.cpp
--- s05/src/faxrecv/recvmain.cpp	2006-08-15 11:13:33.000000000 +0200
+++ s06/src/faxrecv/recvmain.cpp	2012-06-12 20:42:20.799003743 +0200
@@ -136,10 +136,10 @@
     }
 
     if (hLogFile) {
-        fprintf (hLogFile, timebuf);
+        fprintf (hLogFile, "%s", timebuf);
         fprintf (hLogFile, "%s - ", RecvProgName.GetPointer());
         if (statusText) {
-            fprintf (hLogFile, statusText);
+            fprintf (hLogFile, "%s", statusText);
         }
         va_copy(lparams, params);
         vfprintf (hLogFile, text, lparams);
@@ -152,7 +152,7 @@
         char errortext[MAX_STRING_SIZE];
         vsnprintf (errortext, MAX_STRING_SIZE, text, params);
         openlog (DEFAULT_RECVPROG_NAME, LOG_PERROR, 0);
-        syslog (priority, errortext);
+        syslog (priority, "%s", errortext);
         closelog();
     }
 }
diff -urN s05/src/faxsend/faxsend.cpp s06/src/faxsend/faxsend.cpp
--- s05/src/faxsend/faxsend.cpp	2012-06-12 19:54:35.852970513 +0200
+++ s06/src/faxsend/faxsend.cpp	2012-06-12 20:42:23.044522113 +0200
@@ -703,7 +703,7 @@
     sprintf (timebuf + strlen(timebuf), ".%02lu: [%5d]: ", tv.tv_usec / 10000, getpid());
 
     if (hJobFile) {
-        fprintf (hJobFile, timebuf);
+        fprintf (hJobFile, "%s", timebuf);
         vfprintf (hJobFile, text, params);
         va_end (params);
         va_start (params, text);
@@ -742,10 +742,10 @@
     }
 
     if (hLogFile) {
-        fprintf (hLogFile, timebuf);
+        fprintf (hLogFile, "%s", timebuf);
         fprintf (hLogFile, DEFAULT_SENDPROG_NAME " - ");
         if (statusText) {
-            fprintf (hLogFile, statusText);
+            fprintf (hLogFile, "%s", statusText);
         }
         vfprintf (hLogFile, text, params);
         va_end (params);
@@ -756,7 +756,7 @@
         char errortext[MAX_STRING_SIZE];
         vsnprintf (errortext, MAX_STRING_SIZE, text, params);
         openlog (DEFAULT_SENDPROG_NAME, LOG_PERROR, 0);
-        syslog (priority, errortext);
+        syslog (priority, "%s", errortext);
         closelog();
     }
     va_end (params);
diff -urN s05/src/faxsend/sendmain.cpp s06/src/faxsend/sendmain.cpp
--- s05/src/faxsend/sendmain.cpp	2012-06-12 19:54:35.856980361 +0200
+++ s06/src/faxsend/sendmain.cpp	2012-06-12 20:42:25.382267125 +0200
@@ -175,7 +175,7 @@
     ftruncate (fileno (qf), 0);
     while (textQ.Get (&ne) == vTrue) {
         dassert (ne != 0);
-        fprintf (qf, ne);
+        fprintf (qf, "%s", ne);
         delete [] ne;
     }
     fclose (qf);
